### PR TITLE
feat(html/a11y): port noAriaUnsupportedElements to HTML

### PR DESCRIPTION
**Pull request by IxxyDev**
> This PR was written primarily by Claude Code.

## Summary

Ported the `noAriaUnsupportedElements` lint rule from JSX to HTML.

- Biome now detects `role` and `aria-*` attributes on HTML elements that do not support ARIA: `<meta>`, `<html>`, `<script>`, and `<style>`.
- Provides an unsafe fix to remove the offending attribute.
- Added edge-case tests for multiple ARIA attributes on a single element and invalid `aria-*` attributes.

## Test Plan

- [x] `cargo test -p biome_html_analyze -- no_aria_unsupported_elements` — 2 spec tests pass (valid + invalid HTML)
- [x] Invalid cases cover all 4 unsupported elements with both `role` and `aria-*` attributes
- [x] Valid cases confirm that `role`/`aria-*` on supported elements (`<div>`, `<input>`) are not flagged
- [x] Edge case: element with both `role` and `aria-hidden` — only first attribute is reported (consistent with JSX version)
- [x] Edge case: invalid `aria-foo` attribute on `<meta>` — not flagged (consistent with JSX version)

## Docs

Documentation is inline in the rule's rustdoc.

### CODE REVIEW

File: crates/biome_html_analyze/src/lint/a11y/no_aria_unsupported_elements.rs [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+
+        let attributes = element.attributes()?;
+        let attribute = attributes.iter().find_map(|attribute| {
+            let html_attribute = attribute.as_html_attribute()?;
+            let attribute_name = html_attribute.name().ok()?.value_token().ok()?;
+            let attribute_name_text = attribute_name.text_trimmed();
+            (attribute_name_text.starts_with("aria-")
+                || attribute_name_text.eq_ignore_ascii_case("role"))
+            .then_some(attribute)
+        })?;
```
[the remark this answers is not in the document]
Comment (ematipico) [reply crates/biome_html_analyze/src/lint/a11y/no_aria_unsupported_elements.rs:157]: @IxxyDev this is relevant

[the remark this answers is not in the document]
Comment (IxxyDev) [reply crates/biome_html_analyze/src/lint/a11y/no_aria_unsupported_elements.rs:157]: @ematipico adressed + added test cases

File: crates/biome_html_analyze/tests/specs/a11y/noAriaUnsupportedElements/astro/valid.astro [hunk 2]
```diff
@@ -0,0 +1,7 @@
+---
+---
+
+<meta charset="UTF-8" />
+<div role="button"></div>
+<input aria-label="name" />
+<meta charset="UTF-8" aria-foo="bar" />
```
Comment (dyc3) [inline crates/biome_html_analyze/tests/specs/a11y/noAriaUnsupportedElements/astro/valid.astro:7]: We need to make sure that we don't trigger the rule in non-html files for components that have the same names, case insensitively, like `<Meta>`.

File: crates/biome_html_analyze/src/lint/a11y/no_aria_unsupported_elements.rs [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+    fn action(ctx: &RuleContext<Self>, _state: &Self::State) -> Option<HtmlRuleAction> {
+        let element = ctx.query();
+        let mut mutation = ctx.root().begin();
+
+        let attributes = element.attributes()?;
+        let attribute = attributes.iter().find_map(|attribute| {
+            let html_attribute = attribute.as_html_attribute()?;
+            let attribute_name = html_attribute.name().ok()?.value_token().ok()?;
+            let attribute_name_text = attribute_name.text_trimmed();
+            let attribute_name_lower = attribute_name_text.to_lowercase();
```
Comment (dyc3) [inline crates/biome_html_analyze/src/lint/a11y/no_aria_unsupported_elements.rs]: These allocate memory, use `TokenText` instead

File: crates/biome_html_analyze/src/lint/a11y/no_aria_unsupported_elements.rs [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+            .iter()
+            .any(|el| element_name.eq_ignore_ascii_case(el))
+        {
+            let attributes = node.attributes()?;
+            let report = attributes.iter().find_map(|attribute| {
+                let attribute = attribute.as_html_attribute()?;
+                let attribute_name = attribute.name().ok()?.value_token().ok()?;
+                let attribute_name_text = attribute_name.text_trimmed();
+
+                let attribute_name_lower = attribute_name_text.to_lowercase();
```
Comment (dyc3) [inline crates/biome_html_analyze/src/lint/a11y/no_aria_unsupported_elements.rs]: Same thing here it allocates memory

File: crates/biome_html_analyze/tests/specs/a11y/noAriaUnsupportedElements/astro/valid.astro (hunk 2 above)
Comment (IxxyDev) [reply crates/biome_html_analyze/tests/specs/a11y/noAriaUnsupportedElements/astro/valid.astro:7]: fixed

File: crates/biome_html_analyze/src/lint/a11y/no_aria_unsupported_elements.rs (hunk 4 above)
Comment (IxxyDev) [reply crates/biome_html_analyze/src/lint/a11y/no_aria_unsupported_elements.rs]: changed to `TokenText`

File: crates/biome_html_analyze/src/lint/a11y/no_aria_unsupported_elements.rs (hunk 3 above)
Comment (IxxyDev) [reply crates/biome_html_analyze/src/lint/a11y/no_aria_unsupported_elements.rs]: changed to `TokenText`

**Review by dyc3 — APPROVED** [review]
